File: tests/benchmarks/resolution/fixtures/python/repository.py
from models import User


class UserRepository:
    def __init__(self):
        self._store = {}

    def find_by_id(self, user_id):
        return self._store.get(user_id)

    def save(self, user):
        self._store[user.user_id] = user

    def delete(self, user_id):
        return self._store.pop(user_id, None) is not None


def create_repository():
    return UserRepository()

File: tests/benchmarks/resolution/fixtures/rust/repository.rs
use std::collections::HashMap;
use crate::models::{Repository, User};

pub struct UserRepository {
    store: HashMap<u64, User>,
}

impl UserRepository {
    pub fn new() -> Self {
        UserRepository {
            store: HashMap::new(),
        }
    }
}

impl Repository for UserRepository {
    fn find_by_id(&self, id: u64) -> Option<User> {
        self.store.get(&id).cloned()
    }

    fn save(&self, user: &User) -> Result<(), String> {
        // In real code this would take &mut self
        Ok(())
    }

    fn delete(&self, id: u64) -> bool {
        // In real code this would take &mut self
        true
    }
}

pub fn create_repository() -> UserRepository {
    UserRepository::new()
}
